IdentityCore/src/util/mac/MSIDXpcSingleSignOnProvider.m

```diff
@@ -140,39 +146,128 @@ - (void)handleRequestParam:(NSDictionary *)requestParam
+                                                                                     oidcScopes:nil
+                                                                                  correlationId:[NSUUID UUID]
+                                                                                 telemetryApiId:nil
+                                                                            intuneAppIdentifier:nil
+                                                                                    requestType:MSIDRequestBrokeredType
+                                                                                          error:nil];
+        NSError *ssoExtensionRequestError = nil;
+        MSIDSSOExtensionGetDeviceInfoRequest *ssoExtensionRequest = [[MSIDSSOExtensionGetDeviceInfoRequest alloc] initWithRequestParameters:requestParams error:&ssoExtensionRequestError];
+        if (!ssoExtensionRequest || ssoExtensionRequestError)
+        {
+            // This is unlikely to happen, but if it does, return NO
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"[Entra broker] CLIENT get error when creating getDeviceInfoRequest with error: %@", ssoExtensionRequestError);
+            return NO;
+        }
+    
         dispatch_group_t group = dispatch_group_create();
         dispatch_group_enter(group);
-
-        __block BOOL result = NO;
-        MSIDXpcSingleSignOnProvider *localInstance = [MSIDXpcSingleSignOnProvider new];
-        [localInstance getXpcService:^(id<MSIDXpcBrokerInstanceProtocol> xpcService, NSXPCConnection *directConnection, NSError *error) {
-            if (!xpcService || error)
+        // deviceInfo is assigned to the Xpc configuration during json dictionary to model mapping in MSIDDeviceInfo.m
+        [ssoExtensionRequest executeRequestWithCompletion:^(MSIDDeviceInfo * __unused _Nullable deviceInfo, NSError * _Nullable error)
+         {
+            if (error)
             {
+                MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"[Entra broker] CLIENT received deviceInfo with error: %@", error);
                 dispatch_group_leave(group);
                 return;
             }
             
-            NSURL *url = [NSURL URLWithString:MSID_DEFAULT_AAD_AUTHORITY];
-            [xpcService canPerformAuthorization:url completionBlock:^(BOOL canPerformRequest) {
-                [directConnection suspend];
-                [directConnection invalidate];
-                
-                result = canPerformRequest;
-                dispatch_group_leave(group);
-            }];
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"[Entra broker] CLIENT received deviceInfo successfully", nil);
+            dispatch_group_leave(group);
         }];
         
-        // If nothing came back in 2 sec, return false and unblock this thread
-        dispatch_time_t timeout = dispatch_time(DISPATCH_TIME_NOW, 2 * NSEC_PER_SEC);
+        // waiting expired in 1 sec
```

Comment:
Any reason to move from 2 sec to 1 sec?

Reply:
2 sec would be too long for a XPC request to come back. Mostly we don't want to block user for too long to get tokens. However, a more accurate number needs to be provided by using the telemetry dashboard when it get into production